ghost/members-api/lib/repositories/member.js

```diff
@@ -215,6 +215,11 @@ module.exports = class MemberRepository {
             options = {};
         }
 
+        // Member creation is triggered via stripe webhook
+        // Used to ignore free member email alert
+        const viaStripe = options.flag === 'stripe';
```

Comment:
This flag is further coupling Members to Stripe, it would be cool if we didn't have to do that!

In this previous comment https://github.com/TryGhost/Ghost/pull/15315#pullrequestreview-1085464683 I mentioned that it would be nice to use events for this feature, so that we're not adding more dependencies. Now that we're coupling further it seems like a good time to do that 🤔 

As it stands however we're unable to use the event, because it doesn't contain enough information for you to determine whether or not to send the email - that is due to a problem with the architecture of Members, but it shouldn't be too big to solve.

### Solution 1

We can add a `'stripe'` source to the `MemberCreatedEvent` - so that when you catch the event, you can ignore those which were caused by Stripe. This would mean updating the `create` method to somehow allow us to get that source through - perhaps by overloading the `options.context` param, I'm not sure on details - but we'd want it to be generic.

This is a big change though in terms of our data, and I'm not 100% sure we want to add that source to this event, yet at least.

### Solution 2

We can pass a list of subscriptions to the `create` method, and instead of calling `linkSubscription` in the webhook handler, we can call `linkSubscription` in the `create` method, this will mean that the event will be associated with a Member with a `status` of `'paid'` - also allowing you to ignore the event.

Then your handler would look something like this:
```js
handler(event) {
  const member = await getMember(event.data.member_id);
  if (event.data.source !== 'api' || event.data.source !== 'member') {
    return;
  }
  if (member.status === 'free') {
    sendEmail();
  }
}
```

It's a bit of a pain that we only give the `id` in the event for now, but I think we should leave that as is, for now at least.

---

What do you think of splitting this out now - does that make sense to you?

If so, what do you think of these solutions? Can you think of any other ways to reduce the coupling between these two features?

Reply:
Thanks @allouis , as discussed, will rewrite this using Soln 2 as it seems the right time to move this to events.